qubesagent/firewall.py

```diff
@@ -185,10 +185,20 @@ def is_ip6(addr):
 
     def log_error(self, msg):
         self.log.error(msg)
-        subprocess.call(
-            ['notify-send', '-t', '3000', msg],
-            env=os.environ.copy().update({'DISPLAY': ':0'})
-        )
+        try:
+            subprocess.check_output(
+                ['sudo', '-u', self.qdb.read('/default-user') or 'user',
```

Comment:
Maybe use `runuser` instead, to avoid dependence on sudo config (and sudo being installed at all)?

Reply:
I did that. However `runuser` unfortunately requires `DBUS_SESSION_BUS_ADDRESS` on fedora-33, which makes the code pretty ugly (`sudo` seems to do more environment var magic?). Cherry-pick whatever variant you prefer.

It might be worth encapsulating notification code in some Qubes class with screenshot testing etc. long term.